src/Sequence.cc

```diff
@@ -415,14 +416,31 @@ Sequence::execute_if_or_elseif_block(Iterator begin, Iterator end, Context& cont
     return block_end;
 }
 
+bool has_timeout(TimePoint start_time, Timeout timeout)
+{
+    auto timeout_in_ms = static_cast<std::chrono::milliseconds>(timeout).count();
+    return (Clock::now() - start_time).count() > timeout_in_ms;
+}
+
 Sequence::Iterator
 Sequence::execute_range(Iterator step_begin, Iterator step_end, Context& context,
                         CommChannel* comm)
 {
+    using std::chrono::milliseconds;
+    using std::chrono::duration;
+
     Iterator step = step_begin;
+    auto start_time = Clock::now();
```

Comment:
Since we are measuring time differences against a sequence timeout I would encapsulate the functionality into a class instead of using the mutable variables in a sequence or at a Lua hook functionality. The aim is it to have two member functions to control it and hide the mutable part:

```c++
#include "taskolib/Timeout.h"
#include "taskolib/time_types.h"

namespace task {

class TimeoutTrigger {
private:
    Timeout::Duration timeout_{Timeout::infinity()};
    TimePoint start_{Clock::now()};

public:
    void reset(Timeout timeout) { timeout_ = timeout; start_ = Clock::now(); }
    bool is_elapsed() const { return (Clock::now() - start_).count() > timeout_.count(); }
};

} // namespace task
```

What is your meaning about this approach? The code snippet would than change to:

```c++
    ...
    timeout_trigger_.reset(timeout_);

    while(step < step_end)
    {
        if(timeout_trigger_.is_elapsed())
        {
            ...
```

PS: it doesn't have to be with the naming `TimeoutTrigger`. Any other more favourite names is very welcome :-)